File: easy/javascript/0643-maximum-average-subarray.js
/** O(n)
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function (nums, k) {
  let currentSum = 0;
  let topSum = -Infinity;

  for (let idx = 0; idx < k; idx++) {
    currentSum += nums[idx];
    topSum = currentSum;
  }

  for (let idx = k; idx < nums.length; idx++) {
    currentSum -= nums[idx - k];
    currentSum += nums[idx];

    if (currentSum > topSum) {
      topSum = currentSum;
    }
  }

  return topSum / k;
};

File: easy/go/0643-maximum-average-subarray.go
package main

func findMaxAverage(nums []int, k int) float64 {
 currentSum := 0 

 for idx := 0; idx < k; idx ++ {
	currentSum += nums[idx]
 }
 
 topSum := currentSum 

 for idx := k; idx < len(nums); idx++ {
	currentSum -= nums[idx - k]
	currentSum += nums[idx]

	if currentSum > topSum {
		topSum = currentSum
	}
 }

 return float64(topSum) / float64(k)
}